src/corva/handlers.py

```diff
@@ -544,3 +565,43 @@ def _get_custom_event_type_by_raw_aws_event(
         if events:
             return event_type, handler
     return None, None
+
+
+def _merge_events(aws_event: Any, data_transformation_type: Type[RawBaseEvent]) -> Any:
+    """
+    Merges incoming aws_events into one.
+    Merge happens differently, depending on app type.
+    Only "scheduled" and "stream" type of apps can be processed here.
+    If somehow any other type is passed - raise an exception
+    """
+    if data_transformation_type == RawScheduledEvent:
+        # scheduled event
+        if not isinstance(aws_event[0], dict):
+            aws_event = list(itertools.chain(*aws_event))
+        is_depth = aws_event[0]["scheduler_type"] == SchedulerType.data_depth_milestone
+        event_start, event_end = (
+            ("top_depth", "bottom_depth")
+            if is_depth
+            else ("schedule_start", "schedule_end")
+        )
+        min_event_start = min(e[event_start] for e in aws_event)
+        max_event_end = max(
+            (e[event_end] for e in aws_event if e.get(event_end) is not None),
+            default=None,
+        )
+        aws_event[0][event_start] = min_event_start
+        if max_event_end:
+            aws_event[0][event_end] = max_event_end
+        aws_event = aws_event[0]
+        return aws_event
+
+    elif data_transformation_type == RawStreamEvent:
+        # stream event
+        for event in aws_event[1:]:
+            aws_event[0]["records"].extend(event["records"])
+        aws_event = [aws_event[0]]
+        return aws_event
+
+    else:
+        # unexpected event type, raise an exception
+        assert_never(data_transformation_type)  # type: ignore
```

Comment:
`typing_extensions` is not a direct dependency of the sdk, this may be a problem

what about just changing function header? If anyone passes other type - mypy will complain
```python
data_transformation_type: Union[Type[RawScheduledEvent], Type[RawStreamEvent]]
```